lib/rubocop/cop/performance/detect.rb

```diff
@@ -0,0 +1,77 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop is used to identify usages of `select.first` or
+      # `find_all.first` and change them to use `detect` instead.
+      #
+      # @example
+      #   # bad
+      #   [].select { |item| true }.first
+      #   [].find_all { |item| true }.first
+      #
+      #   # good
+      #   [].detect { |item| true }
+      class Detect < Cop
+        MSG = 'Use `%s` instead of `%s.first`.'
+
+        SELECT_METHODS = [:select, :find_all]
+
+        def on_send(node)
+          receiver, second_method = *node
+          return unless second_method == :first
+          return if receiver.nil?
+
+          receiver, _args, _body = *receiver if receiver.block_type?
+
+          _, first_method = *receiver
+          return unless SELECT_METHODS.include?(first_method)
+
+          source_buffer = node.loc.expression.source_buffer
+          location_of_select = receiver.loc.selector.begin_pos
+          end_location = node.loc.selector.end_pos
+
+          range = Parser::Source::Range.new(source_buffer,
```

Comment:
`Range#join` is an easier way to get from the beginning of one range to the end of the next. So, `receiver.loc.selector.join(node.loc.selector)`.